test/decl/func/warn_unused_function_params.swift

```diff
@@ -0,0 +1,45 @@
+// RUN: %target-typecheck-verify-swift -warn-unused-function-params
+
+// SR-2849
+final class FooContainer {
+  func foo1(x: Int) { // expected-warning {{function parameter 'x' was never used; consider replacing it with '_'}} {{13-14=_}}
+    let array = [1, 2, 3]
+    print(array[0])
+  }
+
+  func foo2(_: Int) { // No warning
+    let array = [1, 2, 3]
+    print(array[0])
+  }
+
+  func foo3(_ y: Int) { // // expected-warning {{function parameter 'y' was never used; consider removing it}} {{14-16=}}
+    let array = [1, 2, 3]
+    print(array[0])
+  }
+
+  func foo4(_ y: Int) { // No warning as 'y' is used in the body
+    let array = [1, 2, 3]
+    print(array[y])
+  }
+
+  func foo5(x y: Int) { // expected-warning {{function parameter 'y' was never used; consider replacing it with '_'}} {{15-16=_}}
```

Comment:
Just an opinion here is that `consider replacing it with '_'` to me doesn't see the ideal suggestion and fix-it. Because if you have a parameter in a function that you don't use, the preferable fix would be to remove it. I'm not sure having a function `func foo6(x _: Int)` where a user is forced to pass a parameter that the function wouldn't even use it will be a ideal fix. Just pointing this question out to know if it makes sense and what you guys think :))

Reply:
We could offer another fix-it to remove it entirely

Reply:
I have decided to offer both fix-its. For example, in the original bug report, there was an example of `UITableView` method where ignoring just the parameter name would be the right thing to do, instead of completely ignoring the parameter.

Reply:
Humm I see. Makes sense because the function signature has to match the protocol requirements :)